lib/Mojo/Server/Daemon.pm

```diff
@@ -179,9 +190,12 @@ sub _listen {
   );
 
   return if $self->silent;
+  my $aport
+    = $self->ioloop->acceptor($self->acceptors->[$#{$self->acceptors}])->port;
+  $url->port($aport) if !$url->port;
+  $url->host("*") if !$url->host;
   $self->app->log->info(qq{Listening at "$url"});
   $query->pairs([]);
-  $url->host('127.0.0.1') if $url->host eq '*';
```

Comment:
from a security perspective listening to localhost by default seems safer, but I guess it's also a breaking change that will surprise a lot of people.